nodejs/aws-infra/experimental/ec2.ts

```diff
@@ -0,0 +1,146 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as utils from "../utils";
+
+import * as mod from ".";
+
+export type EC2TaskDefinitionArgs = utils.Overwrite<mod.TaskDefinitionArgs, {
+    /** Not provided.  Defaults automatically to ["EC2"] */
+    requiresCompatibilities?: never;
+
+    /**
+     * Not provided for ec2 task definitions.
+     */
+    cpu?: never;
+    /**
+     * Not provided for ec2 task definitions.
+     */
+    memory?: never;
+
+    /**
+     * Single container to make a ClusterTaskDefinition from.  Useful for simple cases where there
+     * aren't multiple containers, especially when creating a ClusterTaskDefinition to call [run]
+     * on.
+     *
+     * Either [container] or [containers] must be provided.
+     */
+    container?: mod.ContainerDefinition;
+
+    /**
+     * All the containers to make a ClusterTaskDefinition from.  Useful when creating a
+     * ClusterService that will contain many containers within.
+     *
+     * Either [container] or [containers] must be provided.
+     */
+    containers?: Record<string, mod.ContainerDefinition>;
+}>;
+
+export class EC2TaskDefinition extends mod.TaskDefinition {
+    constructor(name: string,
+                args: EC2TaskDefinitionArgs,
+                opts?: pulumi.ComponentResourceOptions) {
+        if (!args.container && !args.containers) {
+            throw new Error("Either [container] or [containers] must be provided");
+        }
+
+        const containers = args.containers || { container: args.container! };
+
+        const argsCopy: mod.TaskDefinitionArgs = {
+            ...args,
+            requiresCompatibilities: ["EC2"],
+            networkMode: pulumi.output(args.networkMode).apply(m => m || "awsvpc"),
+        };
+
+        delete (<any>argsCopy).container;
+        delete (<any>argsCopy).containers;
+
+        super("aws-infra:x:EC2TaskDefinition", name, containers, /*isFargate:*/ false, argsCopy, opts);
+    }
+
+    /**
+     * Creates a service with this as its task definition.
+     */
+    public createService(name: string, args: mod.EC2ServiceArgs, opts?: pulumi.ResourceOptions) {
+        if (args.taskDefinition) {
+            throw new Error("[args.taskDefinition] should not be provided.");
+        }
+
+        if (args.taskDefinitionArgs) {
+            throw new Error("[args.taskDefinitionArgs] should not be provided.");
+        }
+
+        return new mod.EC2Service(name, {
+            ...args,
+            taskDefinition: this,
+        }, opts || { parent: this });
+    }
```

Comment:
i still kept around this instance method.  It seems like a nice way to bridge between resources that are very strongly related.  But we an remove if we don't think this is valuable.